webserver.h

```diff
@@ -11,8 +11,26 @@ using namespace std;
 class WebServer
 {
 private:
+    struct HeaderMiddleware {
+        struct context 
+        {
+        };
+
+        void before_handle(crow::request& req, crow::response& res, context& ctx)
+        {
+            res.manual_length_header = true;
+        }
+
+        void after_handle(crow::request& req, crow::response& res, context& ctx)
+        {
+            res.set_header("Content-Type", "application/json");
+            res.add_header("Access-Control-Allow-Origin", "*");
+            res.add_header("Access-Control-Allow-Methods", "GET, OPTIONS");
+        }
+    };
+
     const vector<unique_ptr<ICanvas>> &_allCanvases; // Reference to all canvases
-    crow::SimpleApp crowApp;
+    crow::App<HeaderMiddleware> _crowApp;
```

Comment:
What's the difference in practical terms?  How did you know to change it?

Reply:
This adds the pre and post request handling processing that's needed to set the response headers that will make NDSCPP's API responses more digestible for certain consumers (read: things like Blazor). 

As to your other question my answer is going to sound like a smart-ass one, but it really isn't meant that way: I read the Crow documentation.